homeassistant/components/smartthings/__init__.py

```diff
@@ -132,9 +133,41 @@ def __init__(self, hass: HomeAssistantType, devices: Iterable,
         """Create a new instance of the DeviceBroker."""
         self._hass = hass
         self._installed_app_id = installed_app_id
+        self.assignments = self._assign_capabilities(devices)
         self.devices = {device.device_id: device for device in devices}
         self.event_handler_disconnect = None
 
+    def _assign_capabilities(self, devices: Iterable):
+        """Assign platforms to capabilities."""
+        assignments = {}
+        for device in devices:
+            capabilities = device.capabilities.copy()
+            slots = {}
+            for platform_name in SUPPORTED_PLATFORMS:
+                platform = importlib.import_module(
+                    '.' + platform_name, self.__module__)
+                assigned = platform.get_capabilities(capabilities)
+                if not assigned:
+                    continue
+                # Draw-down capabilities and set slot assignment
+                for capability in assigned:
+                    if capability not in capabilities:
```

Comment:
We check this in `platform.get_capabilities` too.

Reply:
`get_capabilities` should follow is `Return all capabilities supported if minimum required are present` and I updated the comments to reflect this.  This makes the code cleaner and safer for future platforms.  Sensor and Binary Sensor are slightly different where they only return the intersection, but that works too.